lib/compiler/src/beam_ssa_dead.erl

```diff
@@ -1187,6 +1197,33 @@ opt_redundant_tests_is([I|Is], Tests, Acc) ->
     opt_redundant_tests_is(Is, Tests, [I|Acc]);
 opt_redundant_tests_is([], _Tests, _Acc) -> none.
 
+%% Will any of the variables gain type information from this
+%% operation?
+gains_type_information(#b_set{anno=Anno,op={bif,'=:='},args=Args}) ->
+    Types0 = maps:get(arg_types, Anno, #{}),
+    Types = complete_type_information(Args, 0, Types0),
+    case map_size(Types) of
+        0 ->
+            false;
+        1 ->
+            true;
+        2 ->
+            case Types of
+                #{0 := Same,1 := Same} ->
+                    false;
+                #{} ->
+                    true
```

Comment:
this case was already covered by `map_size/1`, which would fall under the `0` size case.
so it cannot be that `#{}` is reachable, because `map_size(Types)` rules it out. At the same time, when size is 0 we return false, and when size is 0 for #{} we return true. 

I think there is something that may not be correct?